src/jit/codegenarmarch.cpp

```diff
@@ -1554,11 +1554,13 @@ void CodeGen::genCodeForIndir(GenTreeIndir* tree)
     emitAttr    attr       = emitTypeSize(tree);
     instruction ins        = ins_Load(targetType);
 
-    assert((attr != EA_1BYTE) || !(tree->gtFlags & GTF_IND_UNALIGNED));
-
     genConsumeAddress(tree->Addr());
-    if (tree->gtFlags & GTF_IND_VOLATILE)
+    if ((tree->gtFlags & GTF_IND_VOLATILE) != 0)
     {
+        bool aligned = ((tree->gtFlags & GTF_IND_UNALIGNED) == 0);
+
+        assert((attr != EA_1BYTE) || aligned);
```

Comment:
This assert is only necessary because the `useLoadAcquire` formula is missing the term to ignore the GTF_IND_UNALIGNED flag for 1 byte accesses.

I would remove the assert and define 
```
bool aligned = ((tree->gtFlags & GTF_IND_UNALIGNED) == 0) || (attr == EA_1BYTE);
```
and
```
bool useLoadAcquire = genIsValidIntReg(targetReg) && !addr->isContained() &&
    (varTypeIsUnsigned(targetType) || varTypeIsI(targetType)) &&
    aligned;
```